### PR TITLE
Break down components and fixed issue with expired article

### PR DESCRIPTION
**Pull request by yd-fu**
Hi guys,
I did some refactoring majorly restructuring and just breaking down Main page to small components so it's easier to read and looks cleaner. 
Also, I encountered and fixed the issue where list row is empty when the article expired as below:
screen_shot
Plus, I added Lodash for handling array operation. 
please review them  :)

### CODE REVIEW

**Comment by Richard-Cao** [comment]
Awesome!
I will check tonight.

File: app/pages/MainPage/Main.js [hunk 1]
```diff
@@ -83,22 +80,22 @@ class Main extends React.Component {
           pages.push(1);
         });
         store.get('typeList').then(typeList =>
-          this.setState({
-            typeIds,
-            typeList
-          })
-        );
+                    this.setState({
```
Comment (Richard-Cao) [inline app/pages/MainPage/Main.js]: Why add these spaces？

**Comment by Richard-Cao** [comment]
Please update your PR~
Thx~

File: app/pages/MainPage/Main.js (hunk 1 above)
Comment (yd-fu) [inline app/pages/MainPage/Main.js]: ah, i think it's auto-formatting from intellij , I will change it soon , thx :)

**Comment by Richard-Cao** [comment]
Thank you very much~ :)

**Pull request merged**